### DISCUSSION TITLE
0 values with logaritme

### DISCUSSION THREAD

**Discussion by starredev** [carbon-design-system/carbon-charts#1648] [General]
We have rendered a line chart with dates on the bottom axis (scaleType: time), and calculated values on the left axis. When we set the left axis scale type to linear this works as expected and looks fine. However, we need a logarithmic scale on the vertical axis; when we change the left axis scale type to log the chart fails to render and we see the following errors in the browser console.

Receiving few errors

``In log values must be greather than 0``

And

``Error: <path attribute d: Expected number, "<gibberish>...".
Error: <circle> attribute cy: Expected length, "NaN".``

However, for my application I need to use 0 values in a log. Now I'm performing workarounds to use it in this way

````
if(result[1] === 0 && object.transformZeroValues) result[1] = 1;
if(i === lastIndex && object.transformZeroValues) result[1] = 1;
````

![linear](https://github.com/carbon-design-system/carbon-charts/assets/31880091/0e057af6-3b6a-4c9e-a2e7-22aa20ec4dd3)

Actually, this not a great workaround

**Comment by glapadre** [comment]
Did you find a solution to this? I'm currently running into the same issue

**Reply by starredev** [reply]
I had some huge performance issue with carbon actually, and also with this issue I moved over to apexcharts :)
